packages/flame_riverpod/lib/src/consumer.dart

```diff
@@ -139,8 +139,30 @@ mixin RiverpodGameMixin<W extends World> on FlameGame<W> {
   /// Used to facilitate [Component] access to the [ProviderContainer].
   GlobalKey<RiverpodAwareGameWidgetState>? widgetKey;
 
+  final ComponentRef ref = ComponentRef(game: null);
   final List<void Function()> _onBuildCallbacks = [];
 
+  /// Adds a callback method to be invoked in the build method of
+  /// [RiverpodAwareGameWidgetState].
+  void addToGameWidgetBuild(Function() cb) {
+    _onBuildCallbacks.add(cb);
+  }
+
+  @override
+  void onMount() {
+    super.onMount();
+    mounted.whenComplete(() {
+      widgetKey!.currentState!.forceBuild();
+    });
+  }
+
```

Comment:
Can we define onRemove to remove callbacks from the game widget?

Reply:
I think we don’t need this, as dependencies are already disposed in `RiverpodAwareGameWidget.dispose`. And afaik manual removals of callbacks are not supported in ComponentMixin either.

Reply:
> I think we don’t need this, as dependencies are already disposed in `RiverpodAwareGameWidget.dispose`. And afaik manual removals of callbacks are not supported in ComponentMixin either.

The use case that I have in the back of my mind is a swapping out the game associated with a RiverpodGameWidget. In this case I think we would want to remove the various subscriptions associated with the former game, and add the various subscriptions associated with the new game. E.g. imagine that our Flutter app is an implementation of the SNES Classic, which features a small library of different games for the user to select from, with only one being active at any given time. 

In any case, I don't think that we need to ensure that this functionality is supported as of this PR, more of a longer-term goal.